arch/common/shared_irq.ld

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* #if defined(CONFIG_CPU_CORTEX_M_HAS_VTOR) */
+. = ALIGN(4);
+
+_shared_irq_start = .;
+
+KEEP(*(.shared_irq))
+KEEP(*(.shared_irq_arg))
```

Comment:
Still something wrong for me with `CONFIG_CMAKE_LINKER_GENERATOR`.
```
$ west build -b qemu_cortex_m3 tests/arch/arm/arm_irq_advanced_features --pristine -- -DCONFIG_CMAKE_LINKER_GENERATOR=y

...
warning: orphan section `.shared_irq' from `zephyr/CMakeFiles/zephyr_final.dir/isr_tables.c.obj' being placed in section `.shared_irq'
warning: orphan section `.shared_irq_arg' from `zephyr/CMakeFiles/zephyr_final.dir/isr_tables.c.obj' being placed in section `.shared_irq_arg'

```

Reply:
I've made an addition to the PR that works with CONFIG_CMAKE_LINKER_GENERATOR

```
$ west build -b qemu_cortex_m3 tests/arch/arm/arm_irq_advanced_features --pristine -- -DCONFIG_CMAKE_LINKER_GENERATOR=y

...
[150/150] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       20212 B       256 KB      7.71%
             RAM:        5719 B        64 KB      8.73%
        IDT_LIST:          0 GB         2 KB      0.00%
```